arch/arc/core/isr_wrapper.S

```diff
@@ -337,6 +337,13 @@ _skip_sys_power_save_idle_exit:
 SECTION_FUNC(TEXT, _isr_demux)
 	push_s r3
 
+/* according to ARCv2 ISA, r25, r30, r58, r59 are caller-saved
+ * scratch registers, possibly used by interrupt handlers
+ */
+	push r25
+	push r30
+	push r58
```

Comment:
for the accumulator registers (r58/r59), the CONFIG_ARC_HAS_ACCL_REGS should be used/backported as well. Only DSP-flavors of ARC have these.

Reply:
@ruuddw this is a backport PR, is there a patch in master for this one?

Reply:
ARC FPU is not support in 1.14 (supported in PR: #16921), so we can remove the save and restore of this two registers.